Core/Utilities/ElasticSearchDB.py

```diff
@@ -0,0 +1,272 @@
+    return "%s-%s" % ( indexName, today )
+  
+  def createIndex( self, indexPrefix, mapping ):
+    """
+    :param str indexPrefix: it is the index name. 
+    :param dict mapping: the configuration of the index.
+    
+    """
+    result = None
+    fullIndex = self.generateFullIndexName( indexPrefix )  # we have to create the an index in each day...
+    if self.isExists( fullIndex ):
+      result = S_OK( fullIndex )
+    else:
+      try:
+        self.log.info( "Create index: ", fullIndex + str( mapping ) )
+        self.__client.indices.create( fullIndex, body = mapping )
+        result = S_OK( fullIndex )
+      except Exception as e:
+        self.log.error( "Can not create the index:", e )
+        result = S_ERROR( e )
+    return result
+  
+  def deleteIndex( self, indexName ):
+    """
+    :param str indexName the name of the index to be deleted...
+    """
+    try:
+      retVal = self.__client.indices.delete( indexName )
+    except NotFoundError as e:
+      return S_ERROR(e)
+    except ValueError as e:
+      return S_ERROR(e)
+    
+    if retVal.get('acknowledged', False): 
+      return S_OK(indexName)
+    else:
+      return S_ERROR(retVal)
+  
+  def index( self, indexName, doc_type, body ):
+    try:
+      res = self.__client.index( index = indexName, doc_type = doc_type, body = body )
+    except TransportError as e:
+      return S_ERROR(e)
+    
+    if res.get('created', False):
+      return S_OK(indexName)
+    else:
+      return S_ERROR(res)
+    
+  
+  def bulk_index( self, indexprefix, doc_type, data, mapping = {} ):
+    """
+    :param str indexPrefix: it is the index name. 
+    :param str doc_type
+    :param list data contains a list of dictionary 
+    """
+    indexName = self.generateFullIndexName( indexprefix )
+    if not self.isExists( indexName ):
+      retVal = self.createIndex( indexprefix, mapping )
+      if not retVal['OK']:
+        return retVal
+    docs = []
+    for i in data:
+      body = {
+          '_index': indexName,
+          '_type': doc_type,
+          '_source': {}
+      }
+      body['_source'] = i
+      body['_source']['time'] = datetime.fromtimestamp( i.get( 'time', int( Time.toEpoch() ) ) )
+      docs += [body]
+    try:
+      res = helpers.bulk( self.__client, docs, chunk_size = self.__chunk_size )
+    except BulkIndexError as e:
+      return S_ERROR( e )
+    
+    if res[0] == len( docs ):
+      #we have inserted all documents...
+      return S_OK( len( docs ) )
+    else:
+      return S_ERROR( res )
+    return res
+  
+  def getUniqueValue( self, indexName, key, orderBy = False ):
+    """
+    :param str indexName the name of the index which will be used for the query
+    :param dict orderBy it is a dictionary in case we want to order the result {key:'desc'} or {key:'asc'} 
+    It returns a list of unique value for a certain key from the dictionary.
+    """
+    
+    s = self._Search( indexName )
+    if orderBy:
```

Comment:
can't you just call
s.aggs.bucket( key, 'terms', field = key, size = 0, order = orderBy ).metric( key, 'cardinality', field = key )
if orderBy is False, it will not order, or will it ?